gobblin-runtime/src/main/java/gobblin/runtime/local/LocalTaskStateTracker.java

```diff
@@ -93,9 +103,16 @@ public void onTaskCompletion(Task task) {
       LOG.error("Failed to process a task completion callback", t);
     }
 
+    // Add the TaskState of the completed task to the JobState so when the control
+    // returns to the launcher, it sees the TaskStates of all completed tasks.
+    this.jobState.addTaskState(task.getTaskState());
+
     // Mark the completion of this task
     task.markTaskCompletion();
 
+    // Notify the listeners for the completion of the task
+    this.eventBus.post(new NewTaskCompletionEvent(ImmutableList.of(task.getTaskState())));
```

Comment:
There doesn't seem to be any benefit doing it here. You can just post this event once in LocalJobLauncher after all tasks finished.